notificationworkerlambda/src/main/scala/com/gu/notifications/worker/utils/NotificationParser.scala

```diff
@@ -1,18 +1,44 @@
 package com.gu.notifications.worker.utils
 
 import cats.effect.IO
-import models.ShardedNotification
-import play.api.libs.json.{JsError, JsSuccess, Json}
+import com.gu.notifications.worker.tokens.ChunkedTokens
+import models.{ShardRange, ShardedNotification}
+import play.api.libs.json.{Format, JsError, JsSuccess, JsValue, Json}
 
-object NotificationParser {
+case class Event(
+  range: Option[ShardRange],
+  tokens: Option[List[String]]
+)
+
+object Event {
+  implicit val eventJf: Format[Event] = Json.format[Event]
+}
 
-  def parseShardedNotification(input: String): IO[ShardedNotification] = {
-    IO(
-      Json.parse(input).validate[ShardedNotification] match {
-        case JsSuccess(shard, _) => shard
+object NotificationParser {
+  def parseEventNotification(input: String): IO[Either[ShardedNotification, ChunkedTokens]] = {
+    IO {
+      val json: JsValue = Json.parse(input)
+      json.validate[Event] match {
```

Comment:
I think it's fine for now, but we should add a `type` attribute the event upstream to branch on the value of the `type` attribute as a next step.

Reply:
i've swapped to two code paths:
handleShardedNotification and handleChunkedNotification

Currently, handleRequest is supported (for transition purposes) so PROD doesn't need to change any time soon, but will be using the refactored flow for sharded notifications.

This means we don't need `type`

handleEither is there termporarily, but will be removed when second queue and lambda are created